bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitBaseAccessoryHandler.java

```diff
@@ -154,14 +154,15 @@ private void fetchAccessories() {
     protected abstract void accessoriesLoaded();
 
     /**
-     * Extracts the accessory ID from the 'Accessory UID' property.
+     * Returns the accessory ID from the 'AccessoryID' configuration parameter.
      *
      * @return the accessory ID, or null if it cannot be determined
      */
     protected @Nullable Integer getAccessoryId() {
-        if (thing.getProperties().get(PROPERTY_ACCESSORY_UID) instanceof String accessoryUid) {
+        BigDecimal accessoryId = (BigDecimal) getConfig().get(CONFIG_ACCESSORY_ID);
```

Comment:
I don't like this hard cast to BigDecimal .. probably you need to get it as an `Object` and then use `toString()` .. or something like that..